src/main.rs

```diff
@@ -343,13 +343,11 @@ fn run(args: Cli) -> Result<i32> {
     if let Some(ref mut s) = session {
         auth = s.auth()?;
 
-        let mut session_headers = s.headers()?;
-        for key in headers.keys() {
-            session_headers.remove(key);
-        }
-        for (key, value) in session_headers.iter() {
-            headers.append(key, value.clone());
-        }
+        headers = {
+            let mut session_headers = s.headers()?;
+            session_headers.extend(headers);
+            session_headers
+        };
```

Comment:
Don't we still need to remove those keys?

Reply:
I think `.extend()` should take of it. See the example in https://docs.rs/http/latest/http/header/struct.HeaderMap.html#impl-Extend%3C(Option%3CHeaderName%3E,+T)%3E-for-HeaderMap%3CT%3E

```rust
let mut map = HeaderMap::new();

map.insert(ACCEPT, "text/plain".parse().unwrap());
map.insert(HOST, "hello.world".parse().unwrap());

let mut extra = HeaderMap::new();

extra.insert(HOST, "foo.bar".parse().unwrap());
extra.insert(COOKIE, "hello".parse().unwrap());
extra.append(COOKIE, "world".parse().unwrap());

map.extend(extra);

assert_eq!(map["host"], "foo.bar");
assert_eq!(map["accept"], "text/plain");
assert_eq!(map["cookie"], "hello");

let v = map.get_all("host");
assert_eq!(1, v.iter().count());

let v = map.get_all("cookie");
assert_eq!(2, v.iter().count());
```

Reply:
Ahh, got it. I was looking at the wrong Extend impl.